services/airflow/src/dags/datenspende_vitaldata/post_processing/pivot_tables.py

```diff
@@ -0,0 +1,24 @@
+from database import create_db_context, teardown_db_context, execute_sql, query_all_elements
+from psycopg2 import sql
+
+
+def pivot_vitaldata():
+    pivot_targets = [(9, 'steps', 'int'), (65, 'resting_heartrate', 'int')]
+    db_context = create_db_context()
+
+    for vitalid, vitaltype, datatype in pivot_targets:
+        create_pivot_table(db_context, vitalid, vitaltype, datatype)
+
+    teardown_db_context(db_context)
+
+
+def create_pivot_table(db_context, vitalid, vitaltype, datatype):
+    dates = query_all_elements(db_context, "SELECT DISTINCT to_char(date, 'YYYY-MM-DD') from datenspende.vitaldata ORDER BY 1;")
+    dates = [res[0] for res in dates]
+    sql_query = sql.SQL = ("CREATE EXTENSION IF NOT EXISTS tablefunc; DROP TABLE IF EXISTS datenspende.{vitaltype}_ct; CREATE TABLE datenspende.{vitaltype}_ct AS SELECT * FROM crosstab('SELECT user_id, date, value FROM datenspende.vitaldata WHERE type = {vitalid} ORDER BY 1', 'SELECT DISTINCT date from datenspende.vitaldata ORDER BY 1') as columns(user_id int, {columns});".format(
```

Comment:
🤯 I just spent half an hour figuring out the reason for all the downstream tests breaking and here it is.

Reply:
Besides, please move the creation of stored procedures to a database migration and write a test for it there.